lib/Sema/CSSimplify.cpp

```diff
@@ -3699,10 +3701,9 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
 
     return SolutionKind::Solved;
   }
-  
-  
+	
   // If we found some unviable results, then fail, but without recovery.
-  if (!result.UnviableCandidates.empty())
+  if (!result.UnviableCandidates.empty() && !shouldAttemptFixes())
```

Comment:
I think you can remove this check completely because we return `SolutionKind:Error` after `if (shouldAttemptFixes())` block already.